syntax_and_semantics/literals/symbol.md

```diff
@@ -1,45 +1,68 @@
 # Symbol
 
-A [Symbol](http://crystal-lang.org/api/Symbol.html) is a constant that is identified by a name without you having to give it a numeric value.
+A [Symbol](http://crystal-lang.org/api/Symbol.html) represents a unique name inside the entire source code.
+
+Symbols are interpreted at compile time and cannot be created dynamically. The only way to create a Symbol is by using a symbol literal, denoted by a colon (`:`) followed by an identifier. The identifier may be enclosed in double quotes (`"`).
 
 ```crystal
-:hello
-:good_bye
+:unquoted_symbol
+:"quoted symbol"
+:"a" # identical to :a
+```
+
+A double-quoted identifier can contain any unicode character including white spaces and accepts the same escape sequences as a [string literal](./string.html), yet no interpolation.
 
-# With spaces and symbols
-:"symbol with spaces"
+For an unquoted identifier the same naming rules apply as for methods. It can contain alphanumeric characters, underscore (`_`) or characters with a code point greater than `159`(`0x9F`). It must not start with a number and may end with an exclamation mark (`!`) or question mark (`?`).
 
-# Ending with question and exclamation marks
+```crystal
 :question?
 :exclamation!
+```
 
-# For the operators
+All [Crystal operators](../operators.html) can be used as symbol names unquoted:
+```crystal
 :+
 :-
 :*
 :/
-:==
-:<
-:<=
-:>
-:>=
-:!
-:!=
-:=~
-:!~
+:%
 :&
 :|
 :^
-:~
 :**
 :>>
 :<<
-:%
+:==
+:!=
+:<
+:<=
+:>
+:>=
+:<=>
+:===
 :[]
 :[]?
 :[]=
-:<=>
-:===
+:!
+:~
+:!~
+:=~
+```
+
+Internally, symbols are implemented as constants with a numeric value of type `Int32`.
+
+## Percent symbol array literal
+
+Besides the single symbol literal, there is also a percent literal to create an [Array](https://crystal-lang.org/api/Array.html) of symbols. It is indicated by `%i` and a pair of delimiters. Valid delimiters are parenthesis `()`, square brackets `[]`, curly braces `{}`, angles `<>` and pipes `||`. Except for the pipes, all delimiters can be nested meaning a start delimiter inside the string escapes the next end delimiter.
+
+```crystal
+%i(foo bar baz) # => [:foo, :bar, :baz]
+%i(foo\nbar baz) # => [:"foo\nbar", :baz]
+%i(foo(bar) baz) # => [:"foo(bar)", :baz]
```

Comment:
I double check what happen with `%i[a b c(d e)]` and `%i(a b c(d e))` to see how the nesting is working with spaces and different symbols. From the proposed description I thought that was a corner case.

`%i[a b c(d e)] # => [:a, :b, :"c(d", :"e)"]` as expected since the enclosing delimiter and the inner one are not the same, but

`%i(a b c(d e)) => unexpected token: )`

The example `%i(foo(bar) baz)` also complains with `unexpected token: baz`.

Regarding `%i(foo\nbar baz)` the value of the expression is actually `[:"foo\\nbar", :baz]`

I checked with Crystal 0.24.1.

Reply:
These are now all fixed in master.